sdk/cosmos/azure-cosmos/azure/cosmos/diagnostics.py

```diff
@@ -0,0 +1,86 @@
+#The MIT License (MIT)
+#Copyright (c) 2014 Microsoft Corporation
+
+#Permission is hereby granted, free of charge, to any person obtaining a copy
+#of this software and associated documentation files (the "Software"), to deal
+#in the Software without restriction, including without limitation the rights
+#to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#copies of the Software, and to permit persons to whom the Software is
+#furnished to do so, subject to the following conditions:
+
+#The above copyright notice and this permission notice shall be included in all
+#copies or substantial portions of the Software.
+
+#THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#SOFTWARE.
+
+"""Diagnostic tools for Cosmos 
+"""
+
+from requests.structures import CaseInsensitiveDict
+
+class RecordHeaders(object):
```

Comment:
I'd prefer not talking about headers with naming of the object. We will eventually want things like request diagnostics (aka Spans/metrics/etc.)

Biggest issue with this approach is different responses have have different settings. You don't get back quota/etc on readItem calls. You only get back indexing progress on container.read. You only get query metrics on queries. We either have to teach our users all our headers (which we're really not happy about; it's not very discoverable for customers and it doesn't code nicely imo) or build LOTS of these RecordHeaders things.